agent/src/main/java/com/jonnyzzz/teamcity/plugins/node/agent/nvm/NVM.kt

```diff
@@ -105,11 +108,22 @@ class NVMRunner(val downloader : NVMDownloader,
           downloader.downloadNVM(nvmHome, url)
           message("NVM downloaded into $nvmHome")
         }
+        
         script("Install", "Installing Node.js v$version",nvmHome.path) {
-          ". $nvmHome/nvm.sh" n "nvm install $fromSource $version"
+          if (isWindows)
+            installCmd
+          else
+            ". $nvmHome/nvm.sh" n installCmd
         }
+
         script("Use", "Selecting Node.js v$version", nvmHome.path) {
-          ". $nvmHome/nvm.sh" n "nvm use $version" n "eval \${TEAMCITY_CAPTURE_ENV}"
+          if (isWindows)
+            // Executing ${TEAMCITY_CAPTURE_ENV} in Windows has no effect like it does in Linux, as agent parameters
+            // still displays TEAMCITY_CAPTURE_ENV as populated, and its value transfers between build steps without
+            // the need for an 'eval' substitute
+            useCmd
```

Comment:
I have seen Windows symlinks take some time (5-10s sometimes) before they are active, unlike linux symlinks. Using `nvm.exe use` also changes this globally, which could cause problems if more than one process is using node. 

Is there an opportunity to only use nvm-windows to install the node versions, and then prepend the path temporarily with the path? Or set a parameter that is later picked up as the node home folder in other build steps?

NVM installs under `C:\ProgramData\nvm\vX.Y.Z`.

Reply:
I merged it right now. The best approach is to make symlinks be created in somewhere under a build configuration. Using global symlinks is ok, but it may be error prone if there are several build agent running on the same machine. Also, it is the way to allow one build to depend on the output of the next build, because it is too easy to change global machine environment

Reply:
Great. Thanks for merging.

I should have time very soon to look at improving this solution